safe/report/impact_report.py

```diff
@@ -323,11 +323,13 @@ def __init__(
         self._extra_layers = extra_layers
         self._minimum_needs = minimum_needs_profile
         self._multi_exposure_impact_function = multi_exposure_impact_function
-        self._extent = self._iface.mapCanvas().extent()
         self._inasafe_context = InaSAFEReportContext()
 
         # QgsMapSettings is added in 2.4
-        map_settings = self._iface.mapCanvas().mapSettings()
+        if self._iface:
+            map_settings = self._iface.mapCanvas().mapSettings()
+        else:
+            map_settings = QgsMapSettings()
```

Comment:
I am not the best person to review, so perhaps we can ask @timlinux or @Gustry if there is a drawback from doing it.
I hope no :)